client/hooks/use-report-export.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef, useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { recordEvent } from 'wcpay/tracks';
+
+interface ExportResponse {
+	export_id?: string;
+}
+
+interface ExportURLResponse {
+	status: string;
+	download_url?: string;
+}
+
+const maxRetries = 5;
+
+/**
+ * Hook for requesting and polling for a CSV export. E.g. Transactions, Payouts, Disputes.
+ *
+ * @example
+ * const { requestReportExport, isDownloading } = useReportExport();
+ * requestReportExport( {
+ * 	exportRequestURL: '/wc/v3/payments/transactions/download?queryParam=value',
+ * 	exportFileAvailabilityEndpoint: '/wc/v3/payments/transactions/download',
+ * 	userEmail: 'test@example.com',
+ * } );
+ */
+export const useReportExport = () => {
+	const [ isDownloading, setIsDownloading ] = useState( false );
+	const { createNotice } = useDispatch( 'core/notices' );
+	const timeoutIdRef = useRef< NodeJS.Timeout | null >( null );
+	const retryCountRef = useRef( 0 );
+
+	useEffect( () => {
+		// Cancel the timeout if the component unmounts.
+		return () => {
+			if ( timeoutIdRef.current ) {
+				clearTimeout( timeoutIdRef.current );
+			}
+		};
+	}, [ timeoutIdRef ] );
+
+	interface PollForFileProps {
+		/**
+		 * The URL to check for the exported file. E.g. "/wc/v3/payments/transactions/download/{export_id}"
+		 */
+		checkFileURL: string;
+		/**
+		 * The email address the export will be sent to.
+		 */
+		userEmail: string;
+		/**
+		 * The interval in milliseconds to poll the server.
+		 */
+		interval?: number;
+	}
+
+	/**
+	 * Polls the server to check if the exported file is ready.
+	 */
+	async function pollForFile( {
+		checkFileURL,
+		userEmail,
+		interval = 1000,
+	}: PollForFileProps ) {
+		timeoutIdRef.current = setTimeout( async () => {
+			retryCountRef.current++;
+			let exportedFileURLResponse;
+			try {
+				exportedFileURLResponse = await apiFetch< ExportURLResponse >( {
+					path: checkFileURL,
+					method: 'GET',
+				} );
+			} catch ( error ) {
+				// If there's an error (like 500), continue with retry logic
+				exportedFileURLResponse = { status: 'error' };
+			}
+
+			if (
+				'success' === exportedFileURLResponse.status &&
+				exportedFileURLResponse.download_url &&
+				exportedFileURLResponse.download_url.includes(
+					'exports.wordpress.com'
+				)
+			) {
+				// The file is available, so we can download it.
+				// Create a link element to trigger the download.
+				const link = document.createElement( 'a' );
+				// Add force_download=true to the URL to force the download, which adds the appropriate `Content-Disposition: attachment` header when using production server.
+				link.href =
+					exportedFileURLResponse.download_url +
+					'?force_download=true';
+				link.click();
```

Comment:
Will this work 100% reliably in all browsers? A fallback manual download link would be useful if it doesn't work for some reason.

Reply:
This is the existing logic used for downloading the file on this page  (via the library) as well as directly used in  other pages like Dispute Evidence - https://github.com/Automattic/woocommerce-payments/blob/4405c97033b6c3c9b1809bb99415a71d188d1aa7/client/payment-details/dispute-details/evidence-list.tsx#L72 

No issues have been raised so far regarding it having problems on any browser. Recently @Jinksi improved the code with this [change](https://github.com/Automattic/transact-platform-server/pull/7096)

Reply:
I suspect it works well 99.99% of the time, I'm just wondering if a clickable link would be useful/preferable for some users. No biggie, happy to proceed with fully automatic download 👌🏻